### PR TITLE
doc/user: document supported PostgreSQL types

### PR DESCRIPTION
**Pull request by benesch**
And use some fancy CSS to wrap the list to three columns, to avoid taking up too much vertical space in the docs.

image

### Motivation

  * This PR adds a feature requested twice on Slack recently.

### Checklist

- [x] This PR has adequate test coverage / QA involvement has been duly considered.
- [x] This PR has an associated up-to-date [design doc](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/README.md), is a design doc ([template](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/00000000_template.md)), or is sufficiently small to not require a design.
  
- [x] If this PR evolves [an existing `$T ⇔ Proto$T` mapping](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/command-and-response-binary-encoding.md) (possibly in a backwards-incompatible way), then it is tagged with a `T-proto` label.
- [x] If this PR will require changes to cloud orchestration or tests, there is a companion cloud PR to account for those changes that is tagged with the release-blocker label ([example](https://github.com/MaterializeInc/cloud/pull/5021)).
  
- [x] This PR includes the following [user-facing behavior changes](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/guide-changes.md#what-changes-require-a-release-note):
  - n/a

### CODE REVIEW

File: doc/user/content/sql/create-source/postgres.md [hunk 1]
```diff
@@ -152,7 +152,46 @@ ingestion progress and debugging related issues, see [Troubleshooting](/ops/trou
...
+<li><code>float4</code></li>
+<li><code>float8</code></li>
+<li><code>int2</code></li>
+<li><code>int2vector</code></li>
+<li><code>int4</code></li>
+<li><code>int4range</code></li>
+<li><code>int8</code></li>
+<li><code>int8range</code></li>
+<li><code>interval</code></li>
+<li><code>json</code></li>
```
Comment (morsapaes) [inline doc/user/content/sql/create-source/postgres.md:173]: Do we support `json` natively? I thought we were just falling back to `jsonb` under the hood. Just trying to grok how you came to a list including `json` and `timez`, so I know where and how to look in the future. 🙇‍♀️

**Review by morsapaes — APPROVED** [review]

File: doc/user/content/sql/create-source/postgres.md (hunk 1 above)
Comment (benesch) [reply doc/user/content/sql/create-source/postgres.md:219]: I looked here: https://github.com/MaterializeInc/materialize/blob/c8a8ee7c0542d9f40e5afc34ee2606db2d790da9/src/pgrepr/src/types.rs#L546-L654

But I forgot to account for the fact that `timetz` is later converted to an error here: https://github.com/MaterializeInc/materialize/blob/c8a8ee7c0542d9f40e5afc34ee2606db2d790da9/src/pgrepr/src/types.rs#L1004-L1109

We do support `json` natively, but only because of this giant special case: https://github.com/MaterializeInc/materialize/blob/c8a8ee7c0542d9f40e5afc34ee2606db2d790da9/src/sql/src/plan/statement.rs#L781-L786

tl;dr I think the best thing to do is to maintain this list by hand; there's no one place in the codebase that's authoritative, and some of the support/nonsupport is quite subtle. https://github.com/MaterializeInc/materialize/pull/26107 just removed support for the reg* types, for example. PR incoming for that shortly.
